servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/AsyncContext.java

```diff
@@ -582,4 +577,25 @@ private static void disable0() {
         EXECUTOR_PLUGINS.remove(EXECUTOR_PLUGIN);
         LOGGER.info("Disabled. Features that depend on AsyncContext will stop working.");
     }
+
+    private static class CapturedContextProviderUnion implements CapturedContextProvider {
+
+        private final CapturedContextProvider first;
+        private final CapturedContextProvider second;
+
+        CapturedContextProviderUnion(CapturedContextProvider first, CapturedContextProvider second) {
+            this.first = first;
+            this.second = second;
```

Comment:
Should we add `requireNonNull` to make sure both providers are correctly instantiated or this is already guarateed by `ServiceLoader`?

Reply:
I think it's going to be guaranteed, but otoh it doesn't hurt since it should be once on startup.